homeassistant/components/wiz/light.py

```diff
@@ -16,83 +17,177 @@
+        self._attr_min_mireds, self._attr_max_mireds = get_min_max_mireds(bulb_type)
+        self._attr_supported_color_modes = get_supported_color_modes(bulb_type)
+        if supports_effects(bulb_type):
+            self._attr_supported_features = SUPPORT_EFFECT
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, self._light.mac)},
+            name=name,
+            manufacturer="WiZ",
+            model=bulb_type.name,
+        )
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if light is on."""
+        is_on: bool | None = self._light.status
+        return is_on
 
     @property
-    def brightness(self):
+    def brightness(self) -> int | None:
         """Return the brightness of the light."""
-        return self._brightness
+        if (brightness := self._light.state.get_brightness()) is None:
+            return None
+        if 0 <= int(brightness) <= 255:
+            return int(brightness)
+        _LOGGER.error("Received invalid brightness : %s. Expected: 0-255", brightness)
+        return None
 
     @property
-    def rgb_color(self):
-        """Return the color property."""
-        return self._rgb_color
+    def color_mode(self) -> str:
+        """Return the current color mode."""
+        color_modes = self.supported_color_modes
+        assert color_modes is not None
+        if (
+            COLOR_MODE_COLOR_TEMP in color_modes
+            and self._light.state.get_colortemp() is not None
+        ):
+            return COLOR_MODE_COLOR_TEMP
+        if (
+            COLOR_MODE_HS in color_modes
+            and (rgb := self._light.state.get_rgb()) is not None
+            and rgb[0] is not None
+        ):
+            return COLOR_MODE_HS
+        return COLOR_MODE_BRIGHTNESS
 
     @property
-    def hs_color(self):
+    def hs_color(self) -> tuple[float, float] | None:
         """Return the hs color value."""
-        return self._hscolor
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is not None and colortemp != 0:
+            return None
+        if (rgb := self._light.state.get_rgb()) is None:
+            return None
+        if rgb[0] is None:
+            # this is the case if the temperature was changed
+            # do nothing until the RGB color was changed
+            return None
+        if (warmwhite := self._light.state.get_warm_white()) is None:
+            return None
+        hue_sat = convertHSfromRGBCW(rgb, warmwhite)
+        hue: float = hue_sat[0]
+        sat: float = hue_sat[1]
+        return hue, sat
 
     @property
-    def is_on(self):
-        """Return true if light is on."""
-        return self._state
+    def color_temp(self) -> int | None:
+        """Return the CT color value in mireds."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is None or colortemp == 0:
+            return None
+        _LOGGER.debug(
```

Comment:
I don't like that we're adding more logging to state properties. But I assume this is an intermediary refactor.

Reply:
Yes, going away.

Reply:
https://github.com/home-assistant/core/pull/65819